component/grpc/options.go

```diff
@@ -0,0 +1,30 @@
+package grpc
+
+import (
+	"errors"
+
+	"google.golang.org/grpc"
+)
+
+type OptionFunction func(*Component) error
+
+// ServerOptions allows gRPC server options to be set.
+func ServerOptions(options ...grpc.ServerOption) OptionFunction {
+	return func(component *Component) error {
+		if len(options) == 0 {
+			return errors.New("no grpc options provided")
+		}
+
+		component.serverOptions = options
+		return nil
+	}
+}
+
+// Reflection opt-in for gRPC reflection.
+// Reflection could be considered a security risk if services are exposed to public internet.
+func Reflection() OptionFunction {
```

Comment:
This sounds more descriptive indeed yes. If we change though we should probably standardize and change all similar instances in other packages too, to be consistent.

Reply:
In this case we should use the same prefix/pattern in all to be consistent, right?

Reply:
Correct.

Reply:
Maybe we should leave it as is then, or since we are breaking anyway make a decision to do it now.
@wishperera @pkritiotis @xenosdio   WDYT?

Reply:
Sure we can accommodate the change alongside this. So, ideally this would replace option functions with non-boolean input to have a `Use` prefix right? For example `Components(.. params)`will become `UseComponents(..params)` ?

Reply:
Since we are introducing breaking changes, I suggest going all the way through, and not introducing more for the same component as we go.

As for the naming pattern, I would suggest going with the `With` prefix. Personally, I find it makes more sense, especially when many options are present, e.g.
```go
service, _ := patron.New(name, version, patron.WithLogFields(map[string]interface{}{"env": "staging"}), patron.WithRouter(router))
```

But this is opinionated. As long as it is clear, I say we go for it.

Reply:
I agree with the `With` and also if we make the change we should do it in this PR.

Reply:
Yeap, I agree with `With`